exchanges/bybit/bybit_wrapper.go

```diff
@@ -0,0 +1,1844 @@
+					Side:            getSide(openOrders[x].Side),
+					Status:          getOrderStatus(openOrders[x].OrderStatus),
+					Pair:            req.Pairs[i],
+					AssetType:       asset.USDTMarginedFutures,
+					Date:            openOrders[x].CreatedAt,
+				})
+			}
+		}
+	case asset.Futures:
+		for i := range req.Pairs {
+			openOrders, err := by.GetActiveFuturesOrders(ctx, req.Pairs[i], "", "", "", 0)
+			if err != nil {
+				return nil, err
+			}
+
+			for x := range openOrders {
+				orders = append(orders, order.Detail{
+					Price:           openOrders[x].Price,
+					Amount:          openOrders[x].Qty,
+					ExecutedAmount:  openOrders[x].Qty - openOrders[x].LeavesQty,
+					RemainingAmount: openOrders[x].LeavesQty,
+					Fee:             openOrders[x].CumulativeFee,
+					Exchange:        by.Name,
+					OrderID:         openOrders[x].OrderID,
+					ClientOrderID:   openOrders[x].OrderLinkID,
+					Type:            getTradeType(openOrders[x].OrderType),
+					Side:            getSide(openOrders[x].Side),
+					Status:          getOrderStatus(openOrders[x].OrderStatus),
+					Pair:            req.Pairs[i],
+					AssetType:       req.AssetType,
+					Date:            openOrders[x].CreatedAt,
+				})
+			}
+		}
+	case asset.USDCMarginedFutures:
+		openOrders, err := by.GetActiveUSDCOrder(ctx, currency.EMPTYPAIR, "PERPETUAL", "", "", "", "", "", 0)
+		if err != nil {
+			return nil, err
+		}
+
+		for x := range openOrders {
+			for i := range req.Pairs {
+				if req.Pairs[i].String() == openOrders[x].Symbol {
+					orders = append(orders, order.Detail{
+						Price:           openOrders[x].Price,
+						Amount:          openOrders[x].Qty,
+						ExecutedAmount:  openOrders[x].TotalFilledQty,
+						RemainingAmount: openOrders[x].Qty - openOrders[x].TotalFilledQty,
+						Fee:             openOrders[x].TotalFee,
+						Exchange:        by.Name,
+						OrderID:         openOrders[x].ID,
+						ClientOrderID:   openOrders[x].OrderLinkID,
+						Type:            getTradeType(openOrders[x].OrderType),
+						Side:            getSide(openOrders[x].Side),
+						Status:          getOrderStatus(openOrders[x].OrderStatus),
+						Pair:            req.Pairs[i],
+						AssetType:       req.AssetType,
+						Date:            openOrders[x].CreatedAt.Time(),
+					})
+				}
+			}
+		}
+	default:
+		return orders, fmt.Errorf("%s %w", req.AssetType, asset.ErrNotSupported)
+	}
+	order.FilterOrdersByPairs(&orders, req.Pairs)
+	order.FilterOrdersByType(&orders, req.Type)
+	order.FilterOrdersBySide(&orders, req.Side)
+	err := order.FilterOrdersByTimeRange(&orders, req.StartTime, req.EndTime)
+	if err != nil {
+		log.Errorf(log.ExchangeSys, "%s %v", by.Name, err)
+	}
+	return orders, nil
+}
+
+// GetOrderHistory retrieves account order information
+// Can Limit response to specific order status
+func (by *Bybit) GetOrderHistory(ctx context.Context, getOrdersRequest *order.GetOrdersRequest) ([]order.Detail, error) {
```

Comment:
Can link this up to: `GetPastOrders`

Reply:
Okay, 
Can you please also comment if [this](https://bybit-exchange.github.io/docs/inverse/#t-assetexchangerecords) 
can be used for `GetOrderHistory` for the following assets 
`asset.CoinMarginedFutures, asset.USDTMarginedFutures, asset.Futures:`?

@thrasher-

Reply:
That endpoint doesn't appear to have enough data, can use the following:

`USDCPerpetuals`: https://bybit-exchange.github.io/docs/usdc/perpetual/#t-usdcqryorderhistory
`Spot`: https://bybit-exchange.github.io/docs/spot/#t-orderhistory
`USDTMarginedFutures`: https://bybit-exchange.github.io/docs/linear/#t-closedprofitandloss
`CoinMarginedFutures` & `Futures`: https://bybit-exchange.github.io/docs/inverse_futures/#t-closedprofitandloss